core/src/main/java/org/hisp/dhis/android/core/arch/api/executors/internal/CoroutineAPICallExecutorImpl.kt

```diff
@@ -110,7 +110,9 @@ internal class CoroutineAPICallExecutorImpl @Inject constructor(
             val transaction = databaseAdapter.beginNewTransaction()
             try {
                 val result = coroutineScope {
-                    block.invoke()
+                    runBlocking {
```

Comment:
I have surrounded block.invoke() with runBlocking { } to ensure that the successfulTransaction function is only called after all the flows in the block have completed.